lisa/trace.py

```diff
@@ -352,6 +354,69 @@ def ftrace(self):
         """
         return self._ftrace
 
+
+    @classmethod
+    @contextlib.contextmanager
+    def from_target(cls, target, events=[], buffer_size=10240, delete_trace=True, **kwargs):
+        """
+        Context manager that can be used to collect a :class:`Trace` directly
+        from a :class:`lisa.target.Target` without needing to setup an
+        :class:`FtraceCollector`.
+
+        **Example**::
+
+            from lisa.trace import Trace
+            from lisa.target import Target
+
+            target = Target.from_default_conf()
+
+            with Trace.from_target(target, events=['sched_switch', 'sched_wakeup']) as trace:
+                target.execute('echo hello world')
+
+            trace.analysis.tasks.plot_tasks_total_residency(filepath='plot.png')
+
+
+        :param target: Target to connect to.
+        :type target: Target
+
+        :param events: ftrace events to collect and parse in the trace.
+        :type events: list(str)
+
+        :param buffer_size: Size of the ftrace ring buffer.
+        :type buffer_size: int
+
+        :param delete_trace: Delete the trace file on the host once parsed.
+        :type delete_trace: bool
+
+        :Variable keyword arguments: Forwarded to :class:`Trace`.
+        """
+        ftrace_coll = FtraceCollector(target, events=events, buffer_size=buffer_size)
+        plat_info = target.plat_info
+
+        class TraceProxy(TraceBase):
+            def get_view(self, *args, **kwargs):
+                return self.base_trace.get_view(*args, **kwargs)
+
+            def __getattr__(self, attr):
+                return getattr(self.base_trace, attr)
+
+        proxy = TraceProxy()
+
+        with ftrace_coll:
+            yield proxy
+
+        with tempfile.NamedTemporaryFile(suffix='.dat', delete=delete_trace) as _file:
```

Comment:
In `ipynb/examples/typical_experiment.ipynb` we do:
```python
trace_path = os.path.join(wload.res_dir, "trace.dat")
with ftrace_coll:
    wload.run()
ftrace_coll.get_trace(trace_path)
```

So what about letting us specify a result directory? If it's present, save the file there, otherwise go with a tempfile that gets deleted.